pallets/proofs-dealer/src/utils.rs

```diff
@@ -1046,6 +1046,39 @@ impl<T: pallet::Config> ProofsDealerInterface for Pallet<T> {
         )
     }
 
+    // Remove a provider from challenge cycle.
+    fn stop_challenge_cycle(provider_id: &Self::ProviderId) -> DispatchResult {
```

Comment:
This should also be called when the proofs-dealer applies a remove mutation and as a result the forest root would be the default one.

Reply:
Can you point me where this is happening ?

Reply:
The mutation is applied in the `do_submit_proof` function. At some point in the execution we should check that the forest root is the default one, and if it is, we stop the challenge cycle.

Reply:
this is safe to do, since the `do_bsp_confirm_storing` function in the file system pallet will initialize the challenge cycle if the root is no longer the default.

https://github.com/Moonsong-Labs/storage-hub/blob/e722b3c6ca1fae6ee8e4bca31f4154bea69af391/pallets/file-system/src/utils.rs#L1306-L1310

Reply:
Done!